### ISSUE TITLE
Repo name change breaks repo

### ISSUE THREAD

**Issue by GertSchepens** [dwrensha/gitlab-sandstorm#7]
Hi! First of all, great work here! Thanks!
Also, a bit paranoid about being the first to post this :) Vanilla install of Sandstorm on a private machine & on oasis.sandstorm.io

When I change the name of the repo in the settings (Project Settings > project name), the repository breaks. The copy paste data doesn't work anymore. Changing the repo.git to the new name doesn't help either.

gert@Benedikt:~/tmp$  echo url=https://fnepznyl:SV61s9SasuuOz_NV19cB0xzfeAloArzWAMzwC2die9o@api.oasis.sandstorm.io/gitlab/repo.git | git -c credential.helper=store credential approve
gert@Benedikt:~/tmp$  git clone -c credential.helper=store https://fnepznyl@api.oasis.sandstorm.io/gitlab/repo.git repo_fnepznyl_RENAME_ME
Cloning into 'repo_fnepznyl_RENAME_ME'...
fatal: repository 'https://fnepznyl@api.oasis.sandstorm.io/gitlab/repo.git/' not found

![screenshot from 2015-09-09 11 04 12](https://cloud.githubusercontent.com/assets/14195861/9757821/264b6e04-56e3-11e5-85a9-20c602a09e08.png)

**Comment by dwrensha** [comment]
Thanks for reporting this! I'll probably fix it by disabling renaming.

I'm surprised that changing the copy/paste text to the new name doesn't help. I wonder what's going on there.

**Comment by GertSchepens** [comment]
yeah, I'd expected that to work.

Disabling would be a decent fix. & hiding the name on the first page. Slightly redundant with the naming the grain & only one repo per grain. The OCD in me however didn't allow me to go on with "a Wrong Name!!" when the option to change is there :)

**Comment by JamborJan** [comment]
I have the same issue with updated grains which had other names before in an older version of the app which worked perfectly fine.

Now I renamed the repositories back to `repo` and it is working again. 

Maybe it's not only preventing renaming the repos but also making sure they have a valid name when upgrading.

**Comment by jfarlow** [comment]
Same issue still. Renaming the project breaks copy/paste and returns:
"fatal: repository http://-----/repo.git not found"

**Comment by abiggerhammer** [comment]
I'm also having this issue on oasis.sandstorm.io; renaming it breaks it, un-renaming it fixes it.

**Comment by dwrensha** [comment]
As a stopgap, I've disabled renaming in the release that I pushed just now: https://apps.sandstorm.io/app/zx9d3pt0fjh4uqrprjftgpqfwgzp6y2ena6098ug3ctv37uv6kfh

I'm keeping this issue open because I think it would be better to actually support renaming.

**Comment by ntnsndr** [comment]
Agreed, renaming would be extremely useful. Right now mine (on oasis) is stuck on "repo," which is not an arrangement that I'm excited to share with collaborators.

Thanks for all your work in making this possible!

**Comment by memento** [comment]
Hi !
I also think renaming would be useful.
Cheers

**Comment by xet7** [comment]
This issue was moved to sandstormports/gitlab-sandstorm#4
